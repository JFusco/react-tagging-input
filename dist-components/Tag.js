'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Tag = function Tag(props) {
	var onRemoveClick = function onRemoveClick(e) {
		e.preventDefault();

		props.onRemoveTag(e);
	};

	var removeIcon = !props.readOnly ? _react2.default.createElement(
		'a',
		{ onClick: onRemoveClick },
		props.removeTagIcon || String.fromCharCode(215)
	) : null;

	return _react2.default.createElement(
		'li',
		null,
		props.name,
		removeIcon
	);
};

exports.default = Tag;


Tag.propTypes = {
	name: _react2.default.PropTypes.string.isRequired,
	onRemoveTag: _react2.default.PropTypes.func,
	selectedTag: _react2.default.PropTypes.bool,
	readOnly: _react2.default.PropTypes.bool,
	removeTagIcon: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.string, _react2.default.PropTypes.element])
};