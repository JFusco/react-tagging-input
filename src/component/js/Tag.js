// @flow

import React  from 'react';

const Tag = (props: Object) => {
  const onRemoveClick = (e: MouseEvent) => {
    e.preventDefault();

    props.onRemoveTag(e);
  };

  const removeIcon: ?React$Element<any> = !props.readOnly ? (
    <a onClick={onRemoveClick}>
      {props.removeTagIcon|| String.fromCharCode(215)}
    </a>
  ) : null;

  return (
    <li>
      {props.name}
      {removeIcon}
    </li>
  );
};

export default Tag;
