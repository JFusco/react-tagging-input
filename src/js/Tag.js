'use strict';

import React  from 'react';

const Tag = props => {
	const onRemoveClick = e => {
		e.preventDefault();

		props.removeTag();
	};

	const removeIcon = () => {
		if (props.readOnly) return null;

		return (
			<a onClick={onRemoveClick} href="#">
				{props.removeTagIcon}
			</a>
		);
	};

	return (
		<span>
			{/* Tag name */}
			{props.name}

			{/* Tag remove icon */}
			{removeIcon()}
		</span>
	);
};

export default Tag;

Tag.propTypes = {
	name: React.PropTypes.string.isRequired,
	removeTag: React.PropTypes.func.isRequired,
	readOnly: React.PropTypes.bool,
	removeTagIcon: React.PropTypes.oneOfType([
		React.PropTypes.string,
		React.PropTypes.element
	])
};
