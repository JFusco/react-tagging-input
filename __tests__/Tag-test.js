'use strict';

jest.disableAutomock();

import React from 'react';
import { findDOMNode } from 'react-dom';
import { createRenderer, Simulate, renderIntoDocument } from 'react-addons-test-utils';
import Tag from '../src/js/Tag';

const TAG_NAME = 'foo';

describe('Tag', () => {
	let renderer,
		tag;

	beforeEach(() => {
		renderer = createRenderer();

		renderer.render(
			<Tag
				name={TAG_NAME} />
		);

		tag = renderer.getRenderOutput();
	});

	afterEach(() => {
		renderer = null;
		tag = null;
	});

	it('should render', () => {
		expect(tag).not.toBeUndefined();
		expect(tag.type).toBe('li');
	});

	it('should render with a name', () => {
		const props = tag.props.children;

		expect(props[0]).toBe(TAG_NAME);
	});
});


describe('Tag - "readOnly"', () => {
	let renderer;

	beforeEach(() => {
		renderer = createRenderer();
	});

	afterEach(() => {
		renderer = null;
	});

	describe('when readOnly is false', () => {
		it('should render the removeTagIcon', () => {
			renderer.render(
				<Tag
					name={TAG_NAME}
					readOnly={false} />
			);

			const tag = renderer.getRenderOutput();
			const removeIcon = tag.props.children[1];

			expect(removeIcon).not.toBeNull();
			expect(removeIcon.type).toBe('a');
		});
	});

	describe('when readOnly is true', () => {
		it('should not render the removeTagIcon', () => {
			renderer.render(
				<Tag
					name={TAG_NAME}
					readOnly={true} />
			);

			const tag = renderer.getRenderOutput();

			expect(tag.props.children[1]).toBeNull();
		});
	});
});

describe('Tag - "removeTagIcon"', () => {
	let renderer;

	beforeEach(() => {
		renderer = createRenderer();
	});

	afterEach(() => {
		renderer = null;
	});

	describe('when a custom element is passed in', () => {
		it('should render the element', () => {
			const customRemoveIcon = (
				<i className="icon-remove"></i>
			);

			renderer.render(
				<Tag
					name={TAG_NAME}
					removeTagIcon={customRemoveIcon} />
			);

			const tag = renderer.getRenderOutput();
			const removeIcon = tag.props.children[1].props.children;

			expect(tag.props.children[1].type).toBe('a');
			expect(removeIcon.type).toBe('i');
			expect(removeIcon.props.className).toBe('icon-remove');
		});
	});

	describe('when a custom string is passed in', () => {
		it('should render the string', () => {
			const renderer = createRenderer();

			const customRemoveString = 'remove';

			renderer.render(
				<Tag
					name={TAG_NAME}
					removeTagIcon={customRemoveString} />
			);

			const tag = renderer.getRenderOutput();
			const removeIcon = tag.props.children[1].props.children;

			expect(tag.props.children[1].type).toBe('a');
			expect(removeIcon).toBe('remove');
		});
	});
});

describe('Tag - "removeTag"', () => {
	it('should be called when clicking the remove icon', () => {
		const onRemoveClick = jest.genMockFunction();

		const tag = renderIntoDocument(
			<div>
				<Tag
					name={TAG_NAME}
					removeTag={onRemoveClick} />
			</div>
		);

		const statelessTag = findDOMNode(tag).children[0];
		const removeIcon = statelessTag.getElementsByTagName('a')[0];

		Simulate.click(removeIcon);

		expect(onRemoveClick).toBeCalled();
	});
});
