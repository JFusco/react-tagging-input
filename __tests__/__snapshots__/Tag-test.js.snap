// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag - "readOnly" when false should render the removeTagIcon 1`] = `
<li>
  foo
  <a
    onClick={[Function]}
  >
    ×
  </a>
</li>
`;

exports[`Tag - "readOnly" when true should not render the removeTagIcon 1`] = `
<li>
  foo
</li>
`;

exports[`Tag - "removeTagIcon" when a custom element is passed in should render the element 1`] = `
<li>
  foo
  <a
    onClick={[Function]}
  >
    <i
      className="icon-remove"
    />
  </a>
</li>
`;

exports[`Tag - "removeTagIcon" when a custom string is passed in should render the string 1`] = `
<li>
  foo
  <a
    onClick={[Function]}
  >
    remove
  </a>
</li>
`;

exports[`Tag should render 1`] = `
<li>
  foo
  <a
    onClick={[Function]}
  >
    ×
  </a>
</li>
`;
